pkg/controller/daemon/update.go

```diff
@@ -367,12 +368,8 @@ func (dsc *DaemonSetsController) dedupCurHistories(ds *apps.DaemonSet, curHistor
 		}
 		for _, pod := range pods {
 			if pod.Labels[apps.DefaultDaemonSetUniqueLabelKey] != keepCur.Labels[apps.DefaultDaemonSetUniqueLabelKey] {
-				toUpdate := pod.DeepCopy()
-				if toUpdate.Labels == nil {
-					toUpdate.Labels = make(map[string]string)
-				}
-				toUpdate.Labels[apps.DefaultDaemonSetUniqueLabelKey] = keepCur.Labels[apps.DefaultDaemonSetUniqueLabelKey]
-				_, err = dsc.kubeClient.CoreV1().Pods(ds.Namespace).Update(context.TODO(), toUpdate, metav1.UpdateOptions{})
+				patchJson := fmt.Sprintf(`{"metadata": {"labels": {"%s": "%s"}}}`, apps.DefaultDaemonSetUniqueLabelKey, keepCur.Labels[apps.DefaultDaemonSetUniqueLabelKey])
+				_, err = dsc.kubeClient.CoreV1().Pods(ds.Namespace).Patch(context.TODO(), pod.Name, types.StrategicMergePatchType, []byte(patchJson), metav1.PatchOptions{})
```

Comment:
MergePatchType is simpler than StrategicMergePatchType and should cover the functionality this requires